waspc/src/Wasp/AppSpec/App/Auth.hs

```diff
@@ -1,21 +1,61 @@
 {-# LANGUAGE DeriveDataTypeable #-}
+{-# LANGUAGE DuplicateRecordFields #-}
 
 module Wasp.AppSpec.App.Auth
   ( Auth (..),
-    AuthMethod (..),
+    AuthMethods (..),
+    GoogleConfig (..),
+    emailAndPasswordConfig,
+    isEmailAndPasswordAuthEnabled,
+    isGoogleAuthEnabled,
+    isExternalAuthEnabled,
+    onAuthSucceededRedirectToOrDefault,
   )
 where
 
 import Data.Data (Data)
+import Data.Maybe (fromMaybe, isJust)
 import Wasp.AppSpec.Core.Ref (Ref)
 import Wasp.AppSpec.Entity (Entity)
+import Wasp.AppSpec.ExtImport (ExtImport)
 
 data Auth = Auth
   { userEntity :: Ref Entity,
-    methods :: [AuthMethod],
+    methods :: AuthMethods,
     onAuthFailedRedirectTo :: String,
     onAuthSucceededRedirectTo :: Maybe String
   }
   deriving (Show, Eq, Data)
 
-data AuthMethod = EmailAndPassword deriving (Show, Eq, Data)
+data AuthMethods = AuthMethods
+  { emailAndPassword :: Maybe EmailAndPasswordConfig,
+    google :: Maybe GoogleConfig
+  }
+  deriving (Show, Eq, Data)
+
+data EmailAndPasswordConfig = EmailAndPasswordConfig
+  { -- NOTE: Not used right now, but Analyzer does not support an empty data type.
```

Comment:
Indeed, that was how I ended up with this kinda hacky approach/comment. It still needs the deriving stuff since it is used in a type like that above. This is the error you get when empty (i.e. no `configFn` placeholder):

```bash
src/Wasp/Analyzer/StdTypeDefinitions.hs:23:1: error:
    No translation to wasp type for type ConT Wasp.AppSpec.App.Auth.EmailAndPasswordConfig
   |
23 | makeDeclType ''App
   | ^^^^^^^^^^^^^^^^^^
```

It would probably be nice to allow this, but not likely something I'd want to tackle atm :D

Reply:
Interesting that it returned this error, I would expect it to return a bit different type of error! But makes sense, let's leave this for now! Unless you want to play with compiler  a bit, could be fun :D!